plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -0,0 +1,200 @@
+        "{} is not supported by Optuna sweeper.".format(param.type)
+    )
+
+
+def create_optuna_distribution_from_override(override: Override) -> Any:
+    value = override.value()
+    if not override.is_sweep_override():
+        return value
+
+    if override.is_choice_sweep():
+        assert isinstance(value, ChoiceSweep)
+        choices = [x for x in override.sweep_iterator(transformer=Transformer.encode)]
+        _choices = [x for x in choices if isinstance(x, (str, int, float))]
+        assert choices == _choices
+        return CategoricalDistribution(_choices)
+
+    if override.is_range_sweep():
+        assert isinstance(value, RangeSweep)
+        assert value.start is not None
+        assert value.stop is not None
+        if value.shuffle:
+            choices = [
+                x for x in override.sweep_iterator(transformer=Transformer.encode)
+            ]
+            _choices = [x for x in choices if isinstance(x, (str, int, float))]
+            assert choices == _choices
+            return CategoricalDistribution(_choices)
+        return IntUniformDistribution(
+            int(value.start), int(value.stop), step=int(value.step)
+        )
+
+    if override.is_interval_sweep():
+        assert isinstance(value, IntervalSweep)
+        assert value.start is not None
+        assert value.end is not None
+        if "log" in value.tags:
+            if "int" in value.tags:
+                return IntLogUniformDistribution(int(value.start), int(value.end))
+            return LogUniformDistribution(value.start, value.end)
+        else:
+            if "int" in value.tags:
+                return IntUniformDistribution(int(value.start), int(value.end))
+            return UniformDistribution(value.start, value.end)
+
+    raise NotImplementedError("{} is not supported by Optuna sweeper.".format(override))
+
+
+class OptunaSweeperImpl(Sweeper):
+    def __init__(
+        self, optuna_config: OptunaConfig, search_space: Optional[DictConfig]
+    ) -> None:
+        self.optuna_config = optuna_config
+        self.search_space = {}
+        if search_space:
+            assert isinstance(search_space, DictConfig)
+            self.search_space = {
+                x: create_optuna_distribution_from_config(y)
+                for x, y in search_space.items()
+            }
+        self.job_idx: int = 0
+
+    def setup(
+        self,
+        config: DictConfig,
+        config_loader: ConfigLoader,
+        task_function: TaskFunction,
+    ) -> None:
+        self.job_idx = 0
+        self.config = config
+        self.config_loader = config_loader
+        self.launcher = Plugins.instance().instantiate_launcher(
+            config=config, config_loader=config_loader, task_function=task_function
+        )
+        self.sweep_dir = config.hydra.sweep.dir
+
+    def sweep(self, arguments: List[str]) -> None:
+        assert self.config is not None
+        assert self.launcher is not None
+
```

Comment:
```
assert self.job_idx is not None
```
:)

Reply:
Thank you for pointing it out. I added it in https://github.com/facebookresearch/hydra/pull/1132/commits/281de778dc4ecf6f8c153a7f3375936837c074dd.